railties/lib/rails/generators/app_base.rb

```diff
@@ -469,6 +470,58 @@ def run_css
         end
       end
 
+      def confirm_implied_options
+        option_implicators = {
+          active_storage: [:active_record],
+          action_mailbox: [:active_storage],
+          action_text: [:active_storage],
+          hotwire: [:javascript]
+        }
+
+        expand_implied_options!(option_implicators)
+
+        selected_option_implicators = option_implicators.select do |implicator, _|
+          !options["skip_#{implicator}"]
+        end.compact_blank
+
+        skipped_implieds = selected_option_implicators.transform_values do |implieds|
+          implieds.select { |implied| options["skip_#{implied}"] }
+        end.compact_blank
+
+        unless skipped_implieds.empty?
+          implied_options = Hash.new { |h, k| h[k] = [] }
+          skipped_implieds.each do |implicator, implieds|
+            implieds.each { |implied| implied_options[implied] << implicator }
+          end
+
+          say "Based on the specified options, the following options will also be activated:"
+          implied_options.each do |implied, implicators|
+            due_to = implicators.map { |implicator| "--#{implicator.to_s.tr("_", "-")}" }.join(", ")
+            say "* --#{implied.to_s.tr("_", "-")} (due to: #{due_to})"
```

Comment:
This will print out options like `--active-record` and `--javascript` (as in [your screenshot](https://github.com/rails/rails/pull/45223#discussion_r886375551)), but such options either don't exist or [are defined entirely differently](https://github.com/rails/rails/blob/693c14c54d917fc86c9df2e0d6bf1f49d7e6f8ee/railties/lib/rails/generators/rails/app/app_generator.rb#L266).  The options we want to reference are the `--skip-*` options.